homeassistant/components/srp_energy/config_flow.py

```diff
@@ -40,46 +40,54 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Handle a flow initialized by the user."""
-        errors = {}
-        default_title: str = DEFAULT_NAME
-
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
-
-        if self.hass.config.location_name:
-            default_title = self.hass.config.location_name
-
-        if user_input:
-            try:
-                await validate_input(self.hass, user_input)
-            except ValueError:
-                # Thrown when the account id is malformed
-                errors["base"] = "invalid_account"
-            except InvalidAuth:
-                errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                LOGGER.exception("Unexpected exception")
-                return self.async_abort(reason="unknown")
-            else:
-                return self.async_create_entry(title=default_title, data=user_input)
-
+    @callback
+    def _show_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
+        """Show the form to the user."""
+        LOGGER.debug("Show Form")
         return self.async_show_form(
             step_id="user",
             data_schema=vol.Schema(
                 {
                     vol.Required(CONF_ID): str,
                     vol.Required(CONF_USERNAME): str,
                     vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
+                    ): str,
```

Comment:
```suggestion
```
Don't see why we ask for the name, use CONF_ID as the config entry title instead

Reply:
@gjohansson-ST name is used to distinguish between the properties. One could be "Home" and another be "Cabin". It's similar to how weather integrations and other energy integrations are configured.

<img width="554" alt="image" src="https://github.com/home-assistant/core/assets/1304774/12bba491-4dcb-4b85-a7c0-7b3d020a543d">

<img width="322" alt="image" src="https://github.com/home-assistant/core/assets/1304774/c6ab23a0-792c-419b-bd14-2a3334879e65">